build_tools/cmake/iree_macros.cmake

```diff
@@ -123,3 +123,51 @@ function(iree_select_compiler_opts OPTS)
   endif()
   set(${OPTS} ${_OPTS} PARENT_SCOPE)
 endfunction()
+
+#-------------------------------------------------------------------------------
+# Data dependencies
+#-------------------------------------------------------------------------------
+
+# Adds 'data' dependencies to a target.
+#
+# Parameters:
+# NAME: name of the target to add data dependencies to
+# DATA: List of targets and/or files in the source tree. Files should use the
+#       same format as targets (i.e. iree::package::subpackage::file.txt)
+function(iree_data)
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME"
+    "DATA"
+    ${ARGN}
+  )
+
+  if(NOT _RULE_DATA)
+    return()
+  endif()
+
+  foreach(_DATA_LABEL ${_RULE_DATA})
+    if(TARGET ${_DATA_LABEL})
+      add_dependencies(${_RULE_NAME} ${_DATA_LABEL})
+    else()
+      # Not a target, assume to be a file instead.
+      string(REPLACE "::" "/" _FILE_PATH ${_DATA_LABEL})
+
+      # Create a target which copies the data file into the build directory.
+      # If this file is included in multiple rules, only create the target once.
+      string(REPLACE "::" "_" _DATA_TARGET ${_DATA_LABEL})
+      if(NOT TARGET ${_DATA_TARGET})
+        set(_INPUT_PATH "${CMAKE_SOURCE_DIR}/${_FILE_PATH}")
+        set(_OUTPUT_PATH "${CMAKE_BINARY_DIR}/${_FILE_PATH}")
+        add_custom_command(
+          OUTPUT ${_OUTPUT_PATH}
+          COMMAND ${CMAKE_COMMAND} -E copy ${_INPUT_PATH} ${_OUTPUT_PATH}
+        )
+        add_custom_target(${_DATA_TARGET} DEPENDS ${_OUTPUT_PATH})
```

Comment:
Doesn't it depend on its input?

Reply:
(Not sure I fully understand CMake's behavior here)
Indirectly, yes. I have it depending on the output path here, which is generated by the custom command using the input path. I referenced https://gitlab.kitware.com/cmake/cmake/issues/17366 when writing this.

Reply:
Hm, I am not sure if it is really required to call `add_custom_command()` **and** `add_custom_target`. Within `iree_py_library`, which actually tries to achieve the same (copying files to the build directory), I only call `add_custom_target`:

https://github.com/google/iree/blob/9e7cf15e7d1ce7acb2089c567346fdd7a033400c/bindings/python/build_tools/cmake/iree_py_library.cmake#L46-L49

Reply:
👍 that works and is simpler, thanks!